src/Microsoft.ML.StandardLearners/Standard/Online/OnlineGradientDescent.cs

```diff
@@ -46,27 +46,69 @@ public sealed class Arguments : AveragedLinearArguments
             /// </summary>
             public Arguments()
             {
-                LearningRate = (Float)0.1;
-                DecreaseLearningRate = true;
+                LearningRate = OgdDefaultArgs.LearningRate;
+                DecreaseLearningRate = OgdDefaultArgs.DecreaseLearningRate;
             }
+
+            internal class OgdDefaultArgs : AveragedDefaultArgs
+            {
+               internal new const float LearningRate = 0.1f;
+               internal new const bool DecreaseLearningRate = true;
+            }
+        }
+
+        /// <summary>
+        /// Trains a new <see cref="RegressionPredictionTransformer{LinearRegressionPredictor}"/>.
+        /// </summary>
+        /// <param name="env">The pricate instance of <see cref="IHostEnvironment"/>.</param>
+        /// <param name="labelColumn">Name of the label column.</param>
+        /// <param name="featureColumn">Name of the feature column.</param>
+        /// <param name="learningRate">The learning Rate.</param>
+        /// <param name="decreaseLearningRate">Decrease learning rate as iterations progress.</param>
+        /// <param name="l2RegularizerWeight">L2 Regularization Weight.</param>
+        /// <param name="numIterations">Number of training iterations through the data.</param>
+        /// <param name="weightsColumn">The name of the weights column.</param>
+        /// <param name="lossFunction">The custom loss functions. Defaults to <see cref="SquaredLoss"/> if not provided.</param>
+        public OnlineGradientDescentTrainer(IHostEnvironment env,
+            string labelColumn,
```

Comment:
>labelColumn [](start = 19, length = 11)

any reason not to use default values here?

Reply:
i think we have marked the label and feature columns as required everywhere. 

---
In reply to: [221113276](https://github.com/dotnet/machinelearning/pull/1084#discussion_r221113276) [](ancestors = 221113276)

Reply:
I think we were not consistent. 

---
In reply to: [221298419](https://github.com/dotnet/machinelearning/pull/1084#discussion_r221298419) [](ancestors = 221298419,221113276)